src/main/java/pl/edu/icm/pl/mxrdr/extension/importer/pdb/PdbApiImporter.java

```diff
@@ -0,0 +1,80 @@
+package pl.edu.icm.pl.mxrdr.extension.importer.pdb;
+
+import edu.harvard.iq.dataverse.importer.metadata.ImporterData;
+import edu.harvard.iq.dataverse.importer.metadata.ImporterFieldKey;
+import edu.harvard.iq.dataverse.importer.metadata.ImporterFieldType;
+import edu.harvard.iq.dataverse.importer.metadata.ImporterRegistry;
+import edu.harvard.iq.dataverse.importer.metadata.MetadataImporter;
+import edu.harvard.iq.dataverse.importer.metadata.ResultField;
+import pl.edu.icm.pl.mxrdr.extension.importer.MetadataBlock;
+
+import javax.annotation.PostConstruct;
+import javax.ejb.Singleton;
+import javax.ejb.Startup;
+import javax.inject.Inject;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.ResourceBundle;
+
+@Singleton
+@Startup
+public class PdbApiImporter implements MetadataImporter {
+
+    private ImporterRegistry registry;
+    private PdbApiCaller apiCaller;
+    private PdbXmlParser xmlParser;
+
+    // -------------------- CONSTRUCTORS --------------------
+
+    @Deprecated
```

Comment:
Since empty constructor is required by framework and the object wouldn't work if you would use this constructor in unit tests, it is marked like that. I was mainly basing this on docs ```@Deprecated is one that programmers are discouraged from using, typically because it is dangerous, or because a better alternative exists.```

Reply:
By which framework is it required? I recently created a `@Singleton` without a default constructor [here](https://github.com/CeON/mxrdr-extension/blob/master/src/main/java/pl/edu/icm/pl/mxrdr/extension/workflow/MxrdrWorkflowStepSPI.java#L22) and had no issues with it...

Reply:
We are usually using ```@Stateless``` which is from EJB and it requires non-arg constructor. I kinda forgot about it here since I'm CDI anootation but it also requires all classes recursively to be Serializable from what I gathered.